client/components/RoomList.jsx

```diff
@@ -0,0 +1,52 @@
+RoomList = React.createClass({
+
+  goToRoom(roomId, roomName) {
+    this.props.goToRoom(roomId, roomName);
+  },
+
+  goToNewRoom() {
+    this.props.setView("newRoom");
+  },
+
+  render() {
+    var that = this;
+
+    var loader =  (
+      <div className="ui active inverted dimmer">
+        <div className="ui text loader">
+          Loading messages
+        </div>
+      </div>
+    );
+
+    var joinedRooms = this.props.joinedRooms.map(function(room){
+      return <RoomListItem getUserCount={that.props.getUserCount} goToRoom={that.goToRoom} goToNewRoom={that.goToNewRoom} room={room} />;
+    });
+
+    var otherRooms = this.props.otherRooms.map(function(room){
+      return <RoomListItem getUserCount={that.props.getUserCount} goToRoom={that.goToRoom} goToNewRoom={that.goToNewRoom} room={room} />;
+    });
+
+    return (
+      <div className="roomList">
+        <div className="padded">
+          <div className="ui header">
+            Your channels
+          </div>
+          <div className="ui selection list celled">
+            { this.props.subsReady ? joinedRooms : loader}
```

Comment:
Koden blir kortere om du øvest ekstrakter props
`const { subsReady, otherRooms, joinedRooms } = this.props;`